converter.go

```diff
@@ -631,6 +594,184 @@ func arrowToValue(
+			case *array.Int16:
+				v = arrowInt16ToValue(structs.Field(colIdx).(*array.Int16), rowIdx, higherPrecision, fieldMetadata[colIdx].Scale)
+			case *array.Int8:
+				v = arrowInt8ToValue(structs.Field(colIdx).(*array.Int8), rowIdx, higherPrecision, fieldMetadata[colIdx].Scale)
+			}
+		case booleanType:
+			v = arrowBoolToValue(structs.Field(colIdx).(*array.Boolean), rowIdx)
+		case realType:
+			v = arrowRealToValue(structs.Field(colIdx).(*array.Float64), rowIdx)
+		case binaryType:
+			v = arrowBinaryToValue(structs.Field(colIdx).(*array.Binary), rowIdx)
+		case dateType:
+			v = arrowDateToValue(structs.Field(colIdx).(*array.Date32), rowIdx)
+		case timeType:
+			v = arrowTimeToValue(structs.Field(colIdx), rowIdx, fieldMetadata[colIdx].Scale)
+		case textType:
+			v = structs.Field(colIdx).(*array.String).Value(rowIdx)
+		case timestampLtzType, timestampTzType, timestampNtzType:
+			ptr := arrowSnowflakeTimestampToTime(structs.Field(colIdx), getSnowflakeType(fieldMetadata[colIdx].Type), fieldMetadata[colIdx].Scale, rowIdx, loc)
+			v = *ptr
+		case objectType:
+			v = arrowToStructuredType(structs.Field(colIdx).(*array.Struct), fieldMetadata[colIdx].Fields, loc, rowIdx, higherPrecision, params)
+		}
+		m[fieldMetadata[colIdx].Name] = v
+	}
+	return &structuredType{
+		values:        m,
+		fieldMetadata: fieldMetadata,
+		params:        params,
+	}
+}
+
+func arrowDecimal128ToValue(srcValue *array.Decimal128, rowIdx int, higherPrecision bool, scale int) snowflakeValue {
+	if !srcValue.IsNull(rowIdx) {
+		num := srcValue.Value(rowIdx)
+		if scale == 0 {
+			if higherPrecision {
+				return num.BigInt()
+			}
+			return num.ToString(0)
+		}
+		f := decimalToBigFloat(num, int64(scale))
+		if higherPrecision {
+			return f
+		}
+		return fmt.Sprintf("%.*f", scale, f)
+	}
+	return nil
+}
+
+func arrowInt64ToValue(srcValue *array.Int64, rowIdx int, higherPrecision bool, scale int) snowflakeValue {
+	if !srcValue.IsNull(rowIdx) {
+		val := srcValue.Value(rowIdx)
+		if scale == 0 {
+			if higherPrecision {
+				return val
+			}
+			return fmt.Sprintf("%d", val)
+		}
+		if higherPrecision {
+			f := intToBigFloat(val, int64(scale))
+			return f
+		}
+		return fmt.Sprintf("%.*f", scale, float64(val)/math.Pow10(scale))
+	}
+	return nil
+}
+
+func arrowInt32ToValue(srcValue *array.Int32, rowIdx int, higherPrecision bool, scale int) snowflakeValue {
+	if !srcValue.IsNull(rowIdx) {
+		val := srcValue.Value(rowIdx)
+		if scale == 0 {
+			if higherPrecision {
+				return val
+			}
+			return fmt.Sprintf("%d", val)
+		}
+		if higherPrecision {
+			f := intToBigFloat(int64(val), int64(scale))
+			return f
+		}
+		return fmt.Sprintf("%.*f", scale, float64(val)/math.Pow10(scale))
+	}
+	return nil
+}
+
+func arrowInt16ToValue(srcValue *array.Int16, rowIdx int, higherPrecision bool, scale int) snowflakeValue {
+	if !srcValue.IsNull(rowIdx) {
+		val := srcValue.Value(rowIdx)
```

Comment:
I see that there is some duplication among arrowInt**ToValue methods. Perhaps extract/generify this duplicated logic?